packages/kbn-plugin-helpers/src/cli.ts

```diff
@@ -17,59 +17,86 @@
-        flags: command.unknownOptions,
-      }))
-    )
-);
+          build/{plugin.id}-{kibanaVersion}.zip
 
-program
-  .command('build [files...]')
-  .description('Build a distributable archive')
-  .on('--help', docs('build'))
-  .option('--skip-archive', "Don't create the zip file, leave the build path alone")
-  .option(
-    '-d, --build-destination <path>',
-    'Target path for the build output, absolute or relative to the plugin root'
-  )
-  .option('-b, --build-version <version>', 'Version for the build output')
-  .option('-k, --kibana-version <version>', 'Kibana version for the build output')
-  .action(
-    createCommanderAction('build', (command, files) => ({
-      buildDestination: command.buildDestination,
-      buildVersion: command.buildVersion,
-      kibanaVersion: command.kibanaVersion,
-      skipArchive: Boolean(command.skipArchive),
-      files,
-    }))
-  );
+      `,
+      flags: {
+        boolean: ['skip-archive'],
+        string: ['kibana-version'],
+        alias: {
+          k: 'kibana-version',
+        },
+        help: `
+          --skip-archive        Don't create the zip file, just create the build/kibana directory
+          --kibana-version, -v  Kibana version that the
+        `,
+      },
+      async run({ log, flags }) {
+        const versionFlag = flags['kibana-version'];
+        if (versionFlag !== undefined && typeof versionFlag !== 'string') {
+          throw createFlagError('expected a single --kibana-version flag');
+        }
 
-program
-  .command('test:mocha [files...]')
-  .description('Run the server tests using mocha')
-  .on('--help', docs('test/mocha'))
-  .action(
-    createCommanderAction('testMocha', (command, files) => ({
-      files,
-    }))
-  );
+        const skipArchive = flags['skip-archive'];
+        if (skipArchive !== undefined && typeof skipArchive !== 'boolean') {
+          throw createFlagError('expected a single --skip-archive flag');
+        }
 
-program.parse(process.argv);
+        const pluginDir = await findKibanaJson(process.cwd());
+        if (!pluginDir) {
+          throw createFailError(
+            `Unable to find Kibana Platform plugin in [${process.cwd()}] or any of its parent directories. Has it been migrated properly? Does it have a kibana.json file?`
+          );
+        }
+
+        const plugin = loadKibanaPlatformPlugin(pluginDir);
+        const config = await loadConfig(log, plugin);
+        const kibanaVersion = await resolveKibanaVersion(versionFlag, plugin);
+        const sourceDir = plugin.directory;
+        const buildDir = Path.resolve(plugin.directory, 'build/kibana', plugin.manifest.id);
+
+        const context: BuildContext = {
+          log,
+          plugin,
+          config,
+          sourceDir,
+          buildDir,
+          kibanaVersion,
+        };
+
+        await Tasks.initTargets(context);
+        await Tasks.optimize(context);
+        await Tasks.writeServerFiles(context);
```

Comment:
It's not immediately obvious to me why, but with a generated plugin the build is not getting past this point.

```
$ yarn plugin-helpers build --kibana-version 7.6.0
yarn run v1.22.4
$ node ../../scripts/plugin_helpers build --kibana-version 7.6.0
 info deleting the build and target directories
 info running @kbn/optimizer
 │ info initialized, 0 bundles cached
 │ info starting worker [1 bundle]
 │ succ 1 bundles compiled successfully after 5.8 sec
 info copying source into the build and converting with babel
Done in 8.09s.
$
```

Reply:
Turned out the issue was in the `transformFileStream` helper. None of the tests passed in vinyl  directories, which were causing the stream to never complete because of a condition that simply returned instead of calling the bloody callback.